### PR TITLE
[StyleCop] Fix warnings in Text.Number Project

### PR DESCRIPTION
**Pull request by Bill7zz**
- XML comment analysis warning

SpanishNumberRangeParserConfiguration  …
- Move constructor before properties
- Move constructor initializers on their own line
- Add class access modifier

BasePercentageExtractor  …
- Reorder properties a fields
- Set field to a property to keep the variable name with upper-case
- Add returns comments in methods
- Add trailing coma in multi-line initializers
- Use String.Empty instead empty strings

NumberMode  …
- Update comment to fix CS1570 XML comment has badly formed

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs [hunk 1]
```diff
@@ -8,30 +8,28 @@ namespace Microsoft.Recognizers.Text.Number
...
+        protected virtual NumberOptions Options { get; } = NumberOptions.None;
 
         /// <summary>
-        /// extractor the percentage entities from the sentence
+        /// extractor the percentage entities from the sentence.
         /// </summary>
-        /// <param name="source"></param>
-        /// <returns></returns>
+        /// <param name="source">sentnce.</param>
+        /// <returns>List of percentaje entities from the sentence source.</returns>
```
Comment (Aliandi) [inline .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs]: change _percentaje_ to _percentage_

File: .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs [hunk 2]
```diff
@@ -125,13 +124,14 @@ protected static ImmutableHashSet<Regex> BuildRegexes(HashSet<string> regexStrs,
...
         }
 
+        protected abstract ImmutableHashSet<Regex> InitRegexes();
+
         /// <summary>
-        /// replace the @sys.num to the real patterns, directly modifies the ExtractResult
+        /// replace the @sys.num to the real patterns, directly modifies the ExtractResult.
         /// </summary>
-        /// <param name="results">extract results after number extractor</param>
-        /// <param name="originSource">the sentense after replacing the @sys.num, Example: @sys.num %</param>
```
Comment (Aliandi) [inline .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs]: Where did this line go?

File: .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs [hunk 3]
```diff
@@ -200,14 +200,14 @@ protected static ImmutableHashSet<Regex> BuildRegexes(HashSet<string> regexStrs,
...
-        /// <param name="numExtResults"></param>
-        /// <returns></returns>
-        private string PreprocessStrWithNumberExtracted(string str, out Dictionary<int, int> positionMap,
-            out IList<ExtractResult> numExtResults)
+        /// <param name="str">sentence to proccess.</param>
+        /// <param name="positionMap">position Map.</param>
+        /// <param name="numExtResults">number extractor result.</param>
+        /// <returns>return according type "builtin.num" or "builtin.num.percentage".</returns>
+        private string PreprocessStrWithNumberExtracted(
+            string str, out Dictionary<int, int> positionMap, out IList<ExtractResult> numExtResults)
```
Comment (Aliandi) [inline .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs]: Either put them all in the same line like 
` private void PostProcessing(List<ExtractResult> results, string originSource, Dictionary<int, int> positionMap, IList<ExtractResult> numExtResults)`
or one per line
```
 private void PostProcessing(
            List<ExtractResult> results, 
            string originSource, 
            Dictionary<int, int> positionMap,
            IList<ExtractResult> numExtResults)
```

**Review by Aliandi — CHANGES_REQUESTED** [review]

File: .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs [hunk 4]
```diff
@@ -1,295 +1,301 @@
-﻿using System;
```
Comment (JuanAr) [inline .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs]: This file seems fully updated due to changes in line breaks

**Review by JuanAr — CHANGES_REQUESTED** [review]
👍  Everything good except the new lines thing in one file

File: .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs (hunk 4 above)
Comment (Bill7zz) [reply .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs]: Feedback applied.

**Review by JuanAr — APPROVED** [review]

**Review by Aliandi — APPROVED** [review]

**Pull request closed without merging**